spec/unit/jobs/reoccurring_job_spec.rb

```diff
@@ -98,6 +98,148 @@ def perform
         expect(job.polling_interval_seconds).to eq(24.hours)
       end
 
+      context 'exponential backoff rate' do
+        context 'updates the polling interval' do
+          it 'when changing exponential backoff rate only' do
+            TestConfig.config[:broker_client_async_poll_exponential_backoff_rate] = 2.0
+            job = FakeJob.new
+
+            enqueued_time = Time.now
+
+            Jobs::Enqueuer.new(job, queue: Jobs::Queues.generic).enqueue_pollable
+            execute_all_jobs(expected_successes: 1, expected_failures: 0)
+
+            Timecop.freeze(59.seconds.after(enqueued_time)) do
+              execute_all_jobs(expected_successes: 0, expected_failures: 0)
+            end
+
+            Timecop.freeze(61.seconds.after(enqueued_time)) do
+              enqueued_time = Time.now
+              execute_all_jobs(expected_successes: 1, expected_failures: 0)
+            end
+
+            Timecop.freeze(119.seconds.after(enqueued_time)) do
+              execute_all_jobs(expected_successes: 0, expected_failures: 0)
+            end
+
+            Timecop.freeze(121.seconds.after(enqueued_time)) do
+              execute_all_jobs(expected_successes: 1, expected_failures: 0)
+            end
+          end
+          it 'when changing exponential backoff rate and default polling interval' do
+            TestConfig.config[:broker_client_async_poll_exponential_backoff_rate] = 1.3
+            TestConfig.config[:broker_client_default_async_poll_interval_seconds] = 10
+
+            job = FakeJob.new
+
+            enqueued_time = Time.now
+
+            Jobs::Enqueuer.new(job, queue: Jobs::Queues.generic).enqueue_pollable
+            execute_all_jobs(expected_successes: 1, expected_failures: 0)
+
+            Timecop.freeze(9.seconds.after(enqueued_time)) do
+              execute_all_jobs(expected_successes: 0, expected_failures: 0)
+            end
+
+            Timecop.freeze(11.seconds.after(enqueued_time)) do
+              enqueued_time = Time.now
+              execute_all_jobs(expected_successes: 1, expected_failures: 0)
+            end
+
+            Timecop.freeze(12.seconds.after(enqueued_time)) do
+              execute_all_jobs(expected_successes: 0, expected_failures: 0)
+            end
+
+            Timecop.freeze(14.seconds.after(enqueued_time)) do
+              execute_all_jobs(expected_successes: 1, expected_failures: 0)
+            end
+          end
+          it 'when changing exponential backoff rate and retry_after from the job' do
+            job = FakeJob.new(retry_after: ['20', '30'])
```

Comment:
I wonder why the `retry_after` values are strings. Okay, I see that this has been done elsewhere in the code, but then I would rather change it everywhere to numbers.